unittests/test_policies.py

```diff
@@ -115,6 +123,49 @@ def test_system_exit_within_test(self):
         stats = self.runner.stats
         self.assertEqual(1, stats.num_failures())
 
+    def test_retries_bad_check(self):
+        max_retries = 2
+        checks = [BadSetupCheck(system=self.system, resources=self.resources)]
+        self.runner._max_retries = max_retries
+        self.runner.runall(checks, self.system)
+
+        # Ensure that the test was retried #max_retries times and failed.
+        self.assertEqual(1, self.runner.stats.num_cases())
+        self.assertEqual(max_retries, self.runner.stats.current_run)
+        self.assertEqual(1, self.runner.stats.num_failures())
+
+    def test_retries_good_check(self):
+        max_retries = 2
+        checks = [HelloTest(system=self.system, resources=self.resources)]
+        self.runner._max_retries = max_retries
+        self.runner.runall(checks, self.system)
+
+        # Ensure that the test passed without retries.
+        self.assertEqual(1, self.runner.stats.num_cases())
+        self.assertEqual(0, self.runner.stats.current_run)
+        self.assertEqual(0, self.runner.stats.num_failures())
+
+    def test_pass_in_retries(self):
+        max_retries = 3
+        run_to_pass = 2
+        # Create a file containing the current_run; Run 0 will set it to 0,
+        # run 1 to 1 and so on.
+        with tempfile.NamedTemporaryFile(mode='wt', delete=False) as fp:
+            fp.write("-1\n")
```

Comment:
Please use single quotes here. You may also consider my suggestion for the `RetriesCheck` so as to start from the numbering from `0`.

Reply:
Done.